### ISSUE TITLE
window.cmd_focus() and hook.fire("focus_change") do not update border

### ISSUE THREAD

**Issue by surrealoatmeal** [qtile/qtile#3751]
qtile version: `0.21.0`

I am trying to activate a window through a script (the relevant part of which can be viewed [here](https://codeberg.org/surrealcereal/Qmin/src/branch/main/Qmin.py#L164)), but cannot get the window to focus with both the border and the `WindowName` widget updating.

The related keybind in the config below is pressed at the beginning of each video.

Without `hook.fire("focus_change")`, the `WindowName` does not update: [video](https://i.imgur.com/Oo9tzWE.mp4).
With it, it does; but the border does not: [video](https://i.imgur.com/wDwKfgl.mp4).

The barebones config used in this demonstration, with `hook.fire("focus_change")` being commented out for the relevant demo:
```py
from libqtile.config import Screen, Key
from libqtile.bar import Bar
from libqtile import widget, qtile
from libqtile import hook
from libqtile.command import lazy

ID = 27263032

@lazy.function
def demo(qtile):
    window = qtile.windows_map[ID]
    window.cmd_focus()
    hook.fire("focus_change")

if __name__ in ["config", "__main__"]:
    keys = [
        Key(["mod4"], "u",
            demo(),
        ),
        Key(["mod4"], "r",
            lazy.restart(),
        ),
        Key(["mod4"], "Return",
            lazy.spawn("alacritty -e /bin/zsh"),
        ),
        Key(["mod4"], "w",
            lazy.window.kill(),
        ),
    ]
    screens = [
        Screen(top=Bar(widgets=[widget.WindowName()], size=20)),
        Screen(top=Bar(widgets=[widget.WindowName()], size=20))
    ]
```
[The related reddit post and thread with elParaguayo, who instructed me to create this issue.](https://www.reddit.com/r/qtile/comments/wirt0i/move_focus_in_between_windows_gracefully_using_a/)

Edits are me fleshing out the issue.

### Required:

- [X] I have searched past issues to see if this bug has already been reported.

**Comment by elParaguayo** [comment]
I need to take a proper look at this but, from my own test, running `qtile cmd-obj -o window window_id -f focus` did not successfully change the focus (window border was still on the current window, window title widget didn't change).

**Comment by justinesmithies** [comment]
Not sure if this is related @elParaguayo but I'm having issues with this command in my window switcher `client.window[selected_option].focus()` ,the window switcher doesn't switch focus to the window on the same group first time round i have to run it a second time then it goes ?

**Comment by surrealoatmeal** [comment]
What @justinesmithies talked about does not work in my case, multiple triggerings of the `focus()` method does not produce different results.
Video

**Comment by elParaguayo** [comment]
@surrealoatmeal Sorry for the delay. Can you test #4585?

**Comment by elParaguayo** [comment]
Right - think it works now. @surrealoatmeal please test #4585 if you can.

**Comment by surrealoatmeal** [comment]
Hello, sorry for the late response - am checking my mails now.

Unfortunately, I have moved onto greener pastures regarding tiling window managers, and am no longer using qtile. I don't even remember what the original script was doing at this point, so, if it works for the example config I provided above, it's probably also fixed for whatever I was trying to do.

Thanks for your efforts in improving the GNU/Linux tiling window ecosystem, and much love to all other qtile developers.
